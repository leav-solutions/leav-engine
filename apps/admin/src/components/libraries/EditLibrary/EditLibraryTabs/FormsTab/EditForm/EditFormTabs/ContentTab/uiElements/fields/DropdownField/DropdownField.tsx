// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import React from 'react';
import {Dropdown} from 'semantic-ui-react';
import {ICommonFieldsSettings, IFormElementProps} from '../../../_types';

function DropdownField({settings}: IFormElementProps<ICommonFieldsSettings>): JSX.Element {
    return (
        <>
            <label>{settings.label}</label>
            <Dropdown {...settings} />
        </>
    );
}

export default DropdownField;
