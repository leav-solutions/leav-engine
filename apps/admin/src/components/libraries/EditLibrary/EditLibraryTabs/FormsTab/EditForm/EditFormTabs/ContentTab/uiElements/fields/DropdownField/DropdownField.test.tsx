// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {render} from 'enzyme';
import React from 'react';
import DropdownField from './DropdownField';

describe('DropdownField', () => {
    test('Snapshot test', async () => {
        const comp = render(<DropdownField settings={{}} />);

        expect(comp).toMatchSnapshot();
    });
});
