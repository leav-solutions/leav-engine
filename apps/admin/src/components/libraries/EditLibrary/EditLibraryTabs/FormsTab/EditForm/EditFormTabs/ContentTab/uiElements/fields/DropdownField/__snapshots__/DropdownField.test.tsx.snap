// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownField Snapshot test 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": {},
    "children": [],
    "name": "label",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": Node {
      "attribs": {
        "aria-expanded": "false",
        "class": "ui dropdown",
        "role": "listbox",
        "tabindex": "0",
      },
      "children": [
        Node {
          "attribs": {
            "aria-hidden": "true",
            "class": "dropdown icon",
          },
          "children": [],
          "name": "i",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Node {
            "attribs": {
              "class": "menu transition",
            },
            "children": [],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": {
              "class": undefined,
            },
            "x-attribsPrefix": {
              "class": undefined,
            },
          },
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": {
            "aria-hidden": undefined,
            "class": undefined,
          },
          "x-attribsPrefix": {
            "aria-hidden": undefined,
            "class": undefined,
          },
        },
        Node {
          "attribs": {
            "class": "menu transition",
          },
          "children": [],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": Node {
            "attribs": {
              "aria-hidden": "true",
              "class": "dropdown icon",
            },
            "children": [],
            "name": "i",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": {
              "aria-hidden": undefined,
              "class": undefined,
            },
            "x-attribsPrefix": {
              "aria-hidden": undefined,
              "class": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": {
            "class": undefined,
          },
          "x-attribsPrefix": {
            "class": undefined,
          },
        },
      ],
      "name": "div",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": Node {
        "children": [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
      },
      "prev": [Circular],
      "type": "tag",
      "x-attribsNamespace": {
        "aria-expanded": undefined,
        "class": undefined,
        "role": undefined,
        "tabindex": undefined,
      },
      "x-attribsPrefix": {
        "aria-expanded": undefined,
        "class": undefined,
        "role": undefined,
        "tabindex": undefined,
      },
    },
    "parent": Node {
      "children": [
        [Circular],
        Node {
          "attribs": {
            "aria-expanded": "false",
            "class": "ui dropdown",
            "role": "listbox",
            "tabindex": "0",
          },
          "children": [
            Node {
              "attribs": {
                "aria-hidden": "true",
                "class": "dropdown icon",
              },
              "children": [],
              "name": "i",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": {
                  "class": "menu transition",
                },
                "children": [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": {
                  "class": undefined,
                },
                "x-attribsPrefix": {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": {
                "aria-hidden": undefined,
                "class": undefined,
              },
              "x-attribsPrefix": {
                "aria-hidden": undefined,
                "class": undefined,
              },
            },
            Node {
              "attribs": {
                "class": "menu transition",
              },
              "children": [],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": {
                  "aria-hidden": "true",
                  "class": "dropdown icon",
                },
                "children": [],
                "name": "i",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": {
                  "aria-hidden": undefined,
                  "class": undefined,
                },
                "x-attribsPrefix": {
                  "aria-hidden": undefined,
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": {
                "class": undefined,
              },
              "x-attribsPrefix": {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": {
            "aria-expanded": undefined,
            "class": undefined,
            "role": undefined,
            "tabindex": undefined,
          },
          "x-attribsPrefix": {
            "aria-expanded": undefined,
            "class": undefined,
            "role": undefined,
            "tabindex": undefined,
          },
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": {},
    "x-attribsPrefix": {},
  },
  "1": Node {
    "attribs": {
      "aria-expanded": "false",
      "class": "ui dropdown",
      "role": "listbox",
      "tabindex": "0",
    },
    "children": [
      Node {
        "attribs": {
          "aria-hidden": "true",
          "class": "dropdown icon",
        },
        "children": [],
        "name": "i",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": {
            "class": "menu transition",
          },
          "children": [],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": {
            "class": undefined,
          },
          "x-attribsPrefix": {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": {
          "aria-hidden": undefined,
          "class": undefined,
        },
        "x-attribsPrefix": {
          "aria-hidden": undefined,
          "class": undefined,
        },
      },
      Node {
        "attribs": {
          "class": "menu transition",
        },
        "children": [],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": {
            "aria-hidden": "true",
            "class": "dropdown icon",
          },
          "children": [],
          "name": "i",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": {
            "aria-hidden": undefined,
            "class": undefined,
          },
          "x-attribsPrefix": {
            "aria-hidden": undefined,
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": {
          "class": undefined,
        },
        "x-attribsPrefix": {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": [
        Node {
          "attribs": {},
          "children": [],
          "name": "label",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": {},
          "x-attribsPrefix": {},
        },
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": Node {
      "attribs": {},
      "children": [],
      "name": "label",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": [Circular],
      "parent": Node {
        "children": [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
      },
      "prev": null,
      "type": "tag",
      "x-attribsNamespace": {},
      "x-attribsPrefix": {},
    },
    "type": "tag",
    "x-attribsNamespace": {
      "aria-expanded": undefined,
      "class": undefined,
      "role": undefined,
      "tabindex": undefined,
    },
    "x-attribsPrefix": {
      "aria-expanded": undefined,
      "class": undefined,
      "role": undefined,
      "tabindex": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": [
        Node {
          "attribs": {},
          "children": [
            Node {
              "attribs": {},
              "children": [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": {},
                "children": [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": {},
                "x-attribsPrefix": {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": {},
              "x-attribsPrefix": {},
            },
            Node {
              "attribs": {},
              "children": [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": {},
                "children": [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": {},
                "x-attribsPrefix": {},
              },
              "type": "tag",
              "x-attribsNamespace": {},
              "x-attribsPrefix": {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": {},
          "x-attribsPrefix": {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 2,
  "options": {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
